central/networkgraph/flow/datastore/flow_impl.go

```diff
@@ -67,6 +67,10 @@ func (fds *flowDataStoreImpl) GetFlowsForDeployment(ctx context.Context, deploym
 	return flows, nil
 }
 
+func (fds *flowDataStoreImpl) GetExternalFlowsForDeployment(ctx context.Context, deploymentID string) ([]*storage.NetworkFlow, error) {
```

Comment:
I think this function misses access control checks (unfortunately, the network graph flow store is not a generated one, and the logic for result filtering is trickier than the one for common stores).

My take is that the access control would require two steps here:
1) get the target deployment (the flow datastore would need to access the deployment datastore)
```
  elevatedCtx := sac.WithGlobalAccessScopeChecker(
    ctx,
    sac.AllowFixedScopes(
      sac.AcccessModeScopeKeys(storage.Access_READ_ACCESS),
      sac.ResourceScopeKeys(resources.Deployment),
    ),
  )
  deployment, found, err := fds.deploymentDS.GetDeployment(elevatedCtx, deploymentID)
  if err != nil || !found {
    // If the requested source/target deployment does not exist, no need to pull any flow.
    return nil, err
  }
```
2) check the requester has `DeploymentExtension` (network graph) permission to the cluster/namespace where the target deployment is:
```
  allowed, err := networkGraphSAC.ReadAllowed(ctx, sac.KeyForNSScopedObj(deployment))
  if err != nil || !allowed {
    // if the requester is not allowed to see network graph data
    // in the namespace where the request deployment is, return
    // no data.
    return nil, err
  }
```

Both steps should be done before passing the request to the storage layer.

Reply:
I tried to implement these checks at the service level ([here](https://github.com/stackrox/stackrox/pull/12689/files#diff-911ba7c3eb2a2c8dc360db82d65eec89fe9434edcfc7628e6da8c0d305811402R119-R135)) - is that enough or do I need to duplicate/move the checks to this level?